### PR TITLE
33394 sliceviewer cut exports error fixes

### LINKED ISSUE #33394: Error when sliceviewer exports cut as func of spectra in EventWorkspace
**Describe the bug**
Error when sliceviewer exports cut (from line plot or ROI - though the error generated is different) as func of spectra in EventWorkspace (such as CNCS_7860_event.nxs) - but not non-event data such as MAR11060.raw

**To Reproduce**
(1) Load in CNCS_7860_event.nxs
(2) Open in sliceviewer
(3) Enable line plots
(4) Press 'y'
```
Error in execution of algorithm ExtractSpectra:
Invalid data found during construction of HistogramX
ExtractSpectra-v1: Invalid data found during construction of HistogramX
```
(5) Enable ROI
(6) Draw a rectangle
(7) Press 'y'
```
Transpose: EventList: Cannot set Y or E data, these data are generated automatically based on the events
Error in execution of algorithm Transpose:
Transpose: error (see log)
Transpose-v1: Transpose: error (see log)
```

**Platform/Version (please complete the following information):**
 - Found on release-next for 6.3 but bug present in 6.2 release so not a regression
 - 
**Additional context**

### PR DESCRIPTION
**Pull request by warunawickramasingha**
Error when sliceviewer exports Y cuts as func of spectra for EventWorkspace it gave errors on lineplot mode and ROI modes.

#### Summary of work
Error raised in ROI mode was due to trying to apply Transpose algorithm on Event workspaces. As a fix to this Rebin algorithm was applied to PreserveEvents=False for event workspaces. The error raised in lineplot mode was due to having same xvalues in the pixel cuts for event workspaces. This was handled by re-adjusting the xmin and xmax values related to the Y cut which were initially equal to the edges of the corresponding bin.

Fixes #33394. 

### To test:

To Reproduce
(1) Load in CNCS_7860_event.nxs
(2) Open in sliceviewer
(3) Enable line plots
(4) Press 'y' and it should create the Y cut export without any errors
(5) Enable ROI
(6) Draw a rectangle
(7) Press 'y' and should create the Y cut export without any errors

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: qt/python/mantidqt/mantidqt/widgets/sliceviewer/models/roi.py [hunk 1]
```diff
@@ -198,6 +220,25 @@ def _extract_roi_numeric_axis(
     return _extract_region(workspace, xmin, xmax, indexmin, indexmax, roi_name, log_algorithm_calls)
 
 
+def _adjust_xmin_xmax_for_event_workspace(workspace, xmin, xmax, ws_index_min):
```
Comment (robertapplin) [inline qt/python/mantidqt/mantidqt/widgets/sliceviewer/models/roi.py:223]: Would it be possible to add a test for this new function?

**Review by robertapplin — CHANGES_REQUESTED** [review]
The changes look clean, and the error when exporting y cuts no longer happens in lineplot and ROI mode. Just had one suggestion

File: qt/python/mantidqt/mantidqt/widgets/sliceviewer/models/roi.py (hunk 1 above)
Comment (warunawickramasingha) [reply qt/python/mantidqt/mantidqt/widgets/sliceviewer/models/roi.py:223]: Test file added.

**Review by robertapplin — APPROVED** [review]
Thanks for adding the new test file!

**Pull request merged**